SalesforceCommerceCloud/pwa-kit
# @W-14545340@ Add support for node 20

# Description

This PR adds support for node 20/npm 10, mostly by just updating our engines to allow it. Note that **Managed Runtime does not currently support node 20**, so this does not validate deployed apps beyond "can be built with node 20 and executed with node 18".

As discovered in #1535, upgrading to node 20 mitigates _one_ out of memory issue present in node itself. However, after a bit of testing, this did completely mitigate the issue.



# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- (change1)

# How to Test-Drive This PR

To test functionality:

- install node 20 / npm 10
- check out code
- run every npm script in every package
- generate extensible app
- run every npm script in extensible app
- generate non-extensible app
- run every npm script in non-extensible app

To test OOM issue, start the dev server and then execute the following script from `packages/template-retail-react-app`. It updates a file, waits 3 seconds, then makes a request. It repeats until the app crashes or we reach 1000 iterations. Running this locally, I had the app crash after 59 iterations for node 18 and 69 iterations for node 20.
```sh
for i in $(seq 1 1000); do
  echo "// $i" >> app/constants.js
  sleep 3
  code=`curl -sSI -o /dev/null -w "%{http_code}" http://localhost:3000` || break
  echo "$i: $code"
done
```

# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
